// ng-let/esm2020/lib/ng-let.module.mjs
import { NgModule } from '@angular/core';
import { NgLetDirective } from './ng-let.directive';
import * as i0 from "@angular/core";
export class NgLetModule {
}
NgLetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.1", ngImport: i0, type: NgLetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgLetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.3.1", ngImport: i0, type: NgLetModule, declarations: [NgLetDirective], exports: [NgLetDirective] });
NgLetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.3.1", ngImport: i0, type: NgLetModule, providers: [], imports: [[]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.1", ngImport: i0, type: NgLetModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgLetDirective],
                    imports: [],
                    exports: [NgLetDirective],
                    providers: []
                }]
        }] });
